libtransact/src/database/lmdb.rs

```diff
@@ -145,10 +146,13 @@ pub struct LmdbDatabaseReader<'a> {
 }
 
 impl<'a> DatabaseReader for LmdbDatabaseReader<'a> {
-    fn get(&self, key: &[u8]) -> Option<Vec<u8>> {
+    fn get(&self, key: &[u8]) -> Result<Option<Vec<u8>>, DatabaseError> {
         let access = self.txn.access();
-        let val: Result<&[u8], _> = access.get(&self.db.main, key);
-        val.ok().map(Vec::from)
+        access
+            .get(&self.db.main, key)
+            .to_opt()
+            .map(|res: Option<&[u8]>| res.map(Vec::from))
```

Comment:
Should this be `opt` instead of `res`? Same for writer struct.

Reply:
Fixed